python/cudf_polars/cudf_polars/dsl/translate.py

```diff
@@ -246,17 +258,40 @@ def translate_ir(visitor: Any, *, n: int | None = None) -> ir.IR:
         return _translate_ir(node, visitor, schema)
 
 
+def translate_named_expr(visitor: Any, *, n: pl_expr.PyExprIR) -> expr.NamedExpr:
+    """
+    Translate a polars-internal named expression IR object into our representation.
+
+    Parameters
+    ----------
+    visitor
+        Polars NodeTraverser object
+    n
+        Node to translate, a named expression node.
+
+    Returns
+    -------
+    Translated IR object.
+
+    Notes
+    -----
+    The datatype of the internal expression will be obtained from the
+    visitor by calling ``get_dtype``, for this to work properly, the
+    caller should arrange that the expression is translated with the
+    node that it references "active" for the visitor (see :class:`set_node`).
+
+    Raises
+    ------
+    NotImplementedError if any translation fails due to unsupported functionality.
```

Comment:
Following https://numpydoc.readthedocs.io/en/latest/format.html#raises:
```suggestion
    NotImplementedError
        If any translation fails due to unsupported functionality.
```

Reply:
Fixed throughout